From eef1f1f9a01a9c3cf499bd000b2c40a2729cce0c Mon Sep 17 00:00:00 2001
From: Peter Delevoryas <pdel@fb.com>
Date: Tue, 14 Jun 2022 11:32:13 -0700
Subject: [PATCH 506/532] hw: m25p80: Add BP and TB bits to n25q00

This flash model is a 128MB Micron flash model that we can use with fb
machines to test U-Boot.

Signed-off-by: Peter Delevoryas <pdel@fb.com>
---
 hw/block/m25p80.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hw/block/m25p80.c b/hw/block/m25p80.c
index afc3fdf4d6..10cb94b2e7 100644
--- a/hw/block/m25p80.c
+++ b/hw/block/m25p80.c
@@ -267,7 +267,8 @@ static const FlashPartInfo known_devices[] = {
     { INFO("mt25ql512ab", 0x20ba20, 0x1044, 64 << 10, 1024, ER_4K | ER_32K) },
     { INFO_STACKED("mt35xu01g", 0x2c5b1b, 0x104100, 128 << 10, 1024,
                    ER_4K | ER_32K, 2) },
-    { INFO_STACKED("n25q00",    0x20ba21, 0x1000, 64 << 10, 2048, ER_4K, 4) },
+    { INFO_STACKED("n25q00",    0x20ba21, 0x1000, 64 << 10, 2048,
+                   ER_4K | HAS_SR_BP3_BIT6 | HAS_SR_TB, 4) },
     { INFO_STACKED("n25q00a",   0x20bb21, 0x1000, 64 << 10, 2048, ER_4K, 4) },
     { INFO_STACKED("mt25ql01g", 0x20ba21, 0x1040, 64 << 10, 2048, ER_4K, 2) },
     { INFO_STACKED("mt25qu01g", 0x20bb21, 0x1040, 64 << 10, 2048, ER_4K, 2) },
-- 
2.43.0

